mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -1145,6 +1145,46 @@ def Rock_AccelGemmOp:
   let hasVerifier = 1;
 }
 
+// blockwise_broadcasting_reduction
+def Rock_BlockwiseBroadcastReduceOp:
+    Rock_Op<"blockwise_broadcast_reduce">,
+    Arguments<(ins MemRefOf<GemmOutputTypes>:$input,
+                   MemRefOf<LdsBufferTypes>:$workspace_buffer,
+                   MemRefOf<GemmOutputTypes>:$output,
+                   IndexAttr:$axis,
+                   ReduceMethodAttr:$reduceMethod,
+                   TransformMapArrayAttr:$inputRegView,
+                   I32Attr:$blockSize
+                   )> {
+  let summary = "Blockwise reduce operation with output broadcasting";
+  let description = [{
+    The `rock.blockwise_broadcast_reduce` op does reduce at a blockwise level and
+    broadcast the reduce values to a size of input tensor. Note that both input/output
+    tensors live in registers.
```

Comment:
Can we confirm (by staring at ISA etc.) that redundant loads from LDS get eliminated?

Reply:
What redundant loads ?

Reply:
I'm saying that if you load the result values for (i, 0) == (i, 1) ... on a lane (after reducing in j) that you'll only access LDS once

Reply:
Poke re what the codegen looks like

Reply:
I like to propose to defer this exercise to perf improvement stage of this work : https://github.com/ROCmSoftwarePlatform/rocMLIR-internal/issues/988